auparse/sockaddr.go

```diff
@@ -18,10 +18,18 @@
 package auparse
 
 import (
+	"errors"
 	"strconv"
 )
 
+// errInvalidSockAddrSize means socket address size is invalid.
+var errInvalidSockAddrSize = errors.New("invalid socket address size")
```

Comment:
It would be nice if the errors returned included the family and the len of the provided address.
```
type invalidSockAddrErr struct {
    family string
    size   int
}

func (e invalidSockAddrErr) Error() string {
    return fmt.Sprintf("invalid socket address size family=%s: %d, e.family, e.size)
}
```
populated appropriately below.

Reply:
What family to specify in case of `len(sockAddr) < 4`? Or it would be better to introduce another error type for malformed families? Like so:

```go
type invalidSockFamilyError struct {
    family string
}

func (e invalidSockFamilyError) Error() string {
    return fmt.Sprintf("invalid socket address family=%s", e.family)
}

// ...

if len(s) < 4 {
    return nil, invalidSockFamilyError{
        family: s,
    }
}
```

Reply:
Suggest either calling the family "invalid", or conditionally format an error in the `Error` method when the size is less than 4 and don't use the `family` field in that case.
```
type invalidSockAddrErr struct {
    family string
    size   int
}

func (e invalidSockAddrErr) Error() string {
    if e.size < 4 {
        return fmt.Sprintf("invalid family: too short: %d", e.size)
    }
    return fmt.Sprintf("invalid socket address size family=%s: %d, e.family, e.size)
}
```
and in the preamble of the func,
```
func parseSockaddr(s string) (map[string]string, error) {
	if len(s) < 4 {
		return nil, invalidSockAddrErr{size: len(s)}
	}
...
```